src/convertToObject.js

```diff
@@ -3,16 +3,33 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles
+ * (see an example in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const result = {};
+
+  const trimIt = sourceString.trim();
+
+  const splitIt = trimIt.split(';');
+
+  const removeSpaces = splitIt
+    .map(prop => prop.split(':').map(item => item.trim()));
+
+  const removeEmpty = removeSpaces.filter(item => item[0] !== '');
+
+  removeEmpty.forEach((item) => {
+    result[item[0]] = item[1];
+  });
```

Comment:
those methods return the new array, so instead of storing each result to a separate variable, you can just use a chain call instead. So let's do it

Reply:
chain is done